tutorials/Atmos/moisture_and_precip.jl

```diff
@@ -0,0 +1,280 @@
+# This option, coupled with either of the moisture models, is a good start
+# when running in moist `AtmosGCM` configuration.
+# The parameterization of instanteneous precipitation removal is described by the
+# [0-moment microphysics](https://clima.github.io/ClimateMachine.jl/latest/Theory/Atmos/Microphysics_0M/)
+# scheme.
+# The `RainModel` encapsulates the "warm rain" processes, i.e.
+# the formation of precipitation in temperatures above freezing.
+# The `RainSnowModel` describes the full set of microphysical processes
+# leading to the formation of precipitation.
+# Both models are based on
+# [1-moment microphysics](https://clima.github.io/ClimateMachine.jl/latest/Theory/Atmos/Microphysics/)
+# scheme, which is a simple representation of cloud microphysics.
+# It is a good starting point for the moist `AtmosLES` configurations,
+# coupling to the subgrid scale parameterizations, and testing the machine learning pipelines.
+# In the future, a more sophisticated 2-moment microphysics scheme will mostl likely be needed.
+
+
+# # Using CLIMAParameters.jl
+
+# The microphysics parameterizations used to compute the sources and sinks for
+# moisture and precipitation variables assume certain structure
+# in the physical parameters used.
+# This is done in order to leverage the dynamic dispatch mechanism and re-use
+# the internal microphysics functions on different microphysics species
+# based on their type.
+# This structure has to be created when using different parameters from
+# [CLIMAParameters.jl](https://github.com/CliMA/CLIMAParameters.jl).
+
+using CLIMAParameters
+using CLIMAParameters.Planet
+using CLIMAParameters.Atmos.Microphysics
+
+struct LiquidParameterSet <: AbstractLiquidParameterSet end
+struct IceParameterSet <: AbstractIceParameterSet end
+struct RainParameterSet <: AbstractRainParameterSet end
+struct SnowParameterSet <: AbstractSnowParameterSet end
+struct MicropysicsParameterSet{L, I, R, S} <: AbstractMicrophysicsParameterSet
+    liq::L
+    ice::I
+    rai::R
+    sno::S
+end
+struct EarthParameterSet{M} <: AbstractEarthParameterSet
+    microphys::M
+end
+
+const microphys = MicropysicsParameterSet(
+    LiquidParameterSet(),
+    IceParameterSet(),
+    RainParameterSet(),
+    SnowParameterSet(),
+)
+const param_set = EarthParameterSet(microphys)
+
+# After this is done, the created `param_set` should be passed
+# in the same way as in the "dry" tutorials.
+
+
+# # Initial condition
+
+# As discussed in the documentation on the
+# [moisture](https://clima.github.io/ClimateMachine.jl/latest/HowToGuides/Atmos/MoistureModelChoices/)
+# and
+# [precipitation](https://clima.github.io/ClimateMachine.jl/latest/HowToGuides/Atmos/PrecipitationModelChoices/)
+# choices, different model configurations result in different state variables.
+# When using the `EquilMoist` model, the initial condition on total water
+# is the only thing that is needed:
+
+state.moisture.ρq_tot = ρ * 4.2
```

Comment:
I like it! (although 0.0042 is more realistic?)

Reply:
A valid point!

Reply:
Should we move this into docs somewhere and wrap with triple back-ticks? I don't think this will run.

Reply:
@charleskawczynski - Could you be more specific? This is how the initial condition for the squall line experiment is specified:
https://github.com/CliMA/ClimateMachine.jl/blob/6979716190395ee26dabc86ef8195736d66f2117/experiments/AtmosLES/squall_line.jl#L104-L116

If it's just about the multiplication I can just set it to a number.

Reply:
But that's inside a function, which does a bunch of things so that we have this `state.variable` available.

Reply:
[This test](https://github.com/CliMA/ClimateMachine.jl/blob/master/test/Atmos/prog_prim_conversion/runtests.jl) is a relatively small example that allows accessing fields in this sort of way

Reply:
Ah, you mean that the whole tutorial file will not compile? Yeah, probably not, since I'm only talking about certain parts of the setup.

What would be a way to insert code fragments without the need for the whole thing to compile?

Reply:
I do think that this should be in tutorials though. It explains how to augment the experiment setups.

Reply:
> [This test](https://github.com/CliMA/ClimateMachine.jl/blob/master/test/Atmos/prog_prim_conversion/runtests.jl) is a relatively small example that allows accessing fields in this sort of way

I would prefer to avoid adding unneeded lines. I think that already we have too much repetition of code in tutorials. Thats why I wanted to cover here only the lines that matter for the moisture and precipitation.

Reply:
We could `include` a prep file that sets things up, and leave a comment saying something like “for those interested to see more configuration details...”  or something. Aside from that, we could, and have talked about, configuring some sort of harness that allows us to compactly specify things.

alternatively, you could just initialize and use the `AtmosModel`. This is kind of what’s done in the last test in the file I previously pointed to, and requires less boilerplate

Reply:
I think using `include` to add the extra code needed for this to compile is a good idea. If it's in the other tutorials, then I agree there's no need to re-explain it here and it would just bury the important parts you are trying to explain about microphysics.